src/screens/home.tsx

```diff
@@ -14,17 +14,65 @@ const styles = StyleSheet.create({
   text: {
     paddingBottom: 24,
   },
+  dropdown: {
+    width: 200,
+    alignItems: 'center',
+  },
+  dropdownContainer: {
+    flexDirection: 'row',
+  },
 });
-
+type OpenURLButtonProps = {
+  url: string;
+  children: string;
+};
 type HomeScreenProps = ScreenNavigationProps<'Home'>;
 
-const HomeScreen: React.FC<HomeScreenProps> = ({ navigation }) => (
-  <View style={styles.container}>
-    <Text style={styles.text}>Home Screen</Text>
-    <Button mode="contained" onPress={() => navigation.navigate('Details')}>
-      Go to details
-    </Button>
-  </View>
-);
+const HomeScreen: React.FC<HomeScreenProps> = ({ navigation }) => {
+  const [origin, setOrigin] = React.useState('LDN');
+  const [dest, setDest] = React.useState('LDN');
+  const stations = ['SOU', 'RYS', 'OXF', 'RDG', 'WRW'];
+  return (
+    <View style={styles.container}>
+      <View style={styles.dropdownContainer}>
+        <List.Section title="Origin">
+          <List.Accordion title={origin} style={styles.dropdown}>
+            {stations.map((station) => {
+              return (
+                <List.Item
+                  title={station}
+                  key={station}
+                  onPress={() => {
+                    setOrigin(station);
+                  }}
+                />
+              );
+            })}
+          </List.Accordion>
+        </List.Section>
+        <List.Section title="Destination">
+          <List.Accordion title={dest} style={styles.dropdown}>
+            {stations.map((station) => {
+              return (
+                <List.Item
+                  title={station}
+                  key={station}
+                  onPress={() => {
+                    setDest(station);
+                  }}
+                />
+              );
+            })}
+          </List.Accordion>
+        </List.Section>
+      </View>
+      <Text style={styles.text}></Text>
+      <Button onPress={async () => await Linking.openURL(getUrl(dest, origin))}>Plan your journey</Button>
+    </View>
+  );
+};
+function getUrl(dest:string, origin:string):string{
```

Comment:
Generally I'd recommend having the functions above the component - just makes it a little easier to read!